### PR TITLE
TASK: Use `deepObject` for all non trivial parameters and remove `[]` from collection names

### PR DESCRIPTION
**Pull request by mficzel**
Use style deepObject for all QueryParameters that are not scalar or singleValueValueObjects. 

also: 
- added tests for the ValueObjectSpecifications
- improved the ValueObjectSpecifications

### CODE REVIEW

File: Tests/Unit/Domain/Schema/DtoSpecificationTest.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+     */
+    public function testIsSupportedIn(string $className, bool $isSupported, bool $isDTO, bool $isDTC, bool $isSVDTO): void
+    {
+        $this->assertEquals($isSupported, IsSupportedInSchema::isSatisfiedByClassName($className), sprintf('IsSupportedInSchema should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isDTO, IsDataTransferObject::isSatisfiedByClassName($className), sprintf('IsDataTransferObject should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isDTC, IsDataTransferObjectCollection::isSatisfiedByClassName($className), sprintf('IsDataTransferObjectCollection should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isSVDTO, IsSingleValueDataTransferObject::isSatisfiedByClassName($className), sprintf('IsSingleValueDataTransferObject should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+    }
+
+    public function testFoo(): void
```
Comment (nezaniel) [inline Tests/Unit/Domain/Schema/DtoSpecificationTest.php:171]: This is obsolete because already covered in the last test case from above, right?

**Review by nezaniel — CHANGES_REQUESTED** [review]
Looks good, I'd just remove the obsolete test

File: Tests/Unit/Domain/Schema/DtoSpecificationTest.php (hunk 1 above)
Comment (mficzel) [reply Tests/Unit/Domain/Schema/DtoSpecificationTest.php]: Correct. During cleanup this was too far at the end to be noticed.

Comment (mficzel) [reply Tests/Unit/Domain/Schema/DtoSpecificationTest.php]: removed the obsolete test will merge now

**Pull request merged**